api/pkg/apis/v1alpha1/model/summary.go

```diff
@@ -31,11 +32,30 @@ type SummarySpec struct {
 	AllAssignedDeployed bool                        `json:"allAssignedDeployed"`
 }
 type SummaryResult struct {
-	Summary    SummarySpec `json:"summary"`
-	Generation string      `json:"generation"`
-	Time       time.Time   `json:"time"`
+	Summary        SummarySpec  `json:"summary"`
+	Generation     string       `json:"generation"`
+	Time           time.Time    `json:"time"`
+	State          SummaryState `json:"state"`
+	DeploymentHash string       `json:"deploymentHash"`
 }
 
+const (
+	SummaryStatePending SummaryState = iota // Currently unused
+	SummaryStateRunning                     // Should indicate that a reconcile operation is in progress
+	SummaryStateDone                        // Should indicate that a reconcile operation has completed either successfully or unsuccessfully
+)
+
+type SummaryState int
+
 func (s *SummarySpec) UpdateTargetResult(target string, spec TargetResultSpec) {
-	s.TargetResults[target] = spec
+	// If the value in map is not set, set the value; otherwise merge it with the
+	// existing spec
+	if v, ok := s.TargetResults[target]; !ok {
+		s.TargetResults[target] = spec
+	} else {
+		v.Status = spec.Status
+		v.Message = spec.Message
+		maps.Copy(v.ComponentResults, spec.ComponentResults)
```

Comment:
I think this is better than the fix from another PR. We should use this when this PR is merged.

Reply:
Yep, I have dropped this change. Will align with the fix in PR [#225](https://github.com/eclipse-symphony/symphony/pull/225)